rdkit/Chem/RegistrationHash.py

```diff
@@ -0,0 +1,502 @@
+"""
+Generate a unique hash code for a molecule based on chemistry. If two
+molecules are chemically "the same", they should have the same hash.
+
+Used by Schrödinger's LiveDesign to determine if two molecules are
+the same. LiveDesign makes changes to the molecule before molhash,
+somewhat equivalent the steps available in
+rdkit.Chem.MolStandardize.
+
+Using molhash adds value beyond using SMILES because it:
+
+* Ignores SMILES features that are not chemically meaningful
+(e.g. atom map numbers)
+* Canonicalizes enhanced stereochemistry groups. For example
+`C[C@H](O)CC |&1:1|` and `C[C@@H](O)CC |&1:1|` have the same
+molhash
+* Canonicalizes S group data (for example, polymer data)
+
+There are two hash schemes, the default, and one in which
+tautomers are considered equivalent.
+
+
+Copyright (C) 2022 Schrödinger, LLC
+
+"""
+
+import enum
+import hashlib
+import json
+import logging
+import re
+from typing import Iterable
+from typing import Optional
+
+from rdkit import Chem
+from rdkit.Chem import EnumerateStereoisomers
+from rdkit.Chem import rdDepictor
+from rdkit.Chem import rdMolHash
+
+ATOM_PROP_MAP_NUMBER = 'molAtomMapNumber'
+
+logger = logging.getLogger(__name__)
+
+rdDepictor.SetPreferCoordGen(True)
+
+EXTRA_ISOTOPE_REMOVAL_REGEX = re.compile(r'\[[1-3]0*([1-9]?[0-9]*[A-Z][a-z]?)@')
```

Comment:
It would be easier to read/debug the code if these two regexes were defined closer to where they are used